src/util.c

```diff
@@ -36,7 +36,7 @@ char *load_file(const char *path) {
     fseek(file, 0, SEEK_END);
     int length = ftell(file);
     rewind(file);
-    char *data = calloc(length + 1, sizeof(char));
+    char *data = (char*)calloc(length + 1, sizeof(char));
```

Comment:
Going from C to C++ do we need to typecast the alloc's? I'll probably not mess around with it based on what I'm doing, I'm interested in the "why".

Reply:
I think there are several places that I've had to do some casting to make functions C++ compatible. Mostly the functions were trying to return a Null pointer instead of a pointer to whatever we need (i.e. a char in this case).